.yarn/patches/@lavamoat-aa-npm-3.1.2-5a47e9a622.patch

```diff
@@ -0,0 +1,23 @@
+diff --git a/src/index.js b/src/index.js
+index a7e074a776c5e535101442bef16d130bf7eb0dbe..a9d8535e77c020cd9e3171807231454d715ef72b 100644
+--- a/src/index.js
++++ b/src/index.js
+@@ -146,9 +146,18 @@ function processOnePackageInLogicalTree(preferredPackageLogicalPathMap, resolve)
+   return results
+ }
+ 
++const REPLACEMENTS = { 'snaps-utils': '@metamask/snaps-utils' };
```

Comment:
It's possible (I'm not 100% sure) that this would be addressed by adding a "browser" field to some package.jsons that need it. 
Overall, fiddling with the aa algo is not a good idea. WE could introduce support for existing mechanisms that it doesn't support yet. It does support the browser field AFAIR, but doesn't support symlinks. 

I'd need to understand why the special treatment was necessary to begin with before I can come up with anything more concrete.

Reply:
Our tsconfig specifies paths that allows us to reference the monorepo packages by their package name (e.g. `@metamask/snaps-utils`), but during build time these will be replaced with relative paths: https://github.com/MetaMask/snaps-monorepo/blob/main/tsconfig.packages.json#L11

This confuses LavaMoats naming scheme

Reply:
In the bundle all references of `@metamask/snaps-utils` are replaced with `../snaps-utils/src/index`. 

This causes LavaMoat to name the index of that package as: `external:../snaps-utils/index.ts`. The index then requires `./caveats` and LavaMoat throws `required package not in allowlist: package "external:../snaps-utils/src/index.ts" requested "external:../snaps-utils/src/caveats.ts" as "./caveats"`

Reply:
Seems that this has fixed itself after I started using another plugin for our tsconfig paths. Now all the paths in question are using `external:` without LavaMoat complaining. Another patch removed! 🎉